.github/actions/export-branch/action.yml

```diff
@@ -0,0 +1,13 @@
+name: 'Export BRANCH variable'
+description: 'Makes BRANCH variable accessible to jobs'
+runs:
+  using: "composite"
+  steps:
+    - name: Export BRANCH variable
+      shell: bash
+      run: |
+        export BRANCH=${GITHUB_REF##*/}
```

Comment:
> If the branch is directly available, and is typically only used by a single step in the workflow (last step in this case), then why bother with an action instead of just export BRANCH=${GITHUB_REF##*/} inside the run: |?

You're right, in that case we don't need to define an action.

> why do you use ##*/? Wouldn't this work the same: export BRANCH=${GITHUB_REF}?

Generally GITHUB_REF is set as /ref/head/branch-name 
```
$ export GITHUB_REF=refs/head/foobar
$ echo ${GITHUB_REF##*/}
foobar
```

> GITHUB_REF is defined as "The branch or tag ref that triggered the workflow". I wonder if this will confuse any of the scripts.

I will look at older travis runs and see what was the value of BRANCH being evaluated there and make sure that we set the BRANCH variable to the same value.

Reply:
Updated the PR removing action and doing things in workflow itself.

1) Setting BRANCH env as ${{ github.head_ref }} at job level which is set only in pull requests (https://docs.github.com/en/free-pro-team@latest/actions/reference/context-and-expression-syntax-for-github-actions#github-context)
2) Later setting BRANCH env if BRANCH is not already set (push to master) from GITHUB_REF.